src/workerd/api/base64.c++

```diff
@@ -6,35 +6,38 @@
 
 namespace workerd::api {
 
-kj::Array<kj::byte> Base64Module::decodeArray(kj::Array<kj::byte> input) {
-  auto size = simdutf::maximal_binary_length_from_base64((const char*)input.begin(), input.size());
-  auto buf = kj::heapArray<kj::byte>(size);
-  auto result = simdutf::base64_to_binary(
-      input.asChars().begin(), input.size(), buf.asChars().begin(), simdutf::base64_default);
+jsg::BufferSource Base64Module::decodeArray(jsg::Lock& js, jsg::BufferSource input) {
+  auto ptr = input.asArrayPtr();
+  auto size = simdutf::maximal_binary_length_from_base64((const char*)ptr.begin(), ptr.size());
```

Comment:
```suggestion
  auto size = simdutf::maximal_binary_length_from_base64(static_cast<const char*>(ptr.begin()), ptr.size());
```

nit: What's our convention on C-style casts? I tend to avoid them.

Reply:
eh, probably best to avoid but I kept it here since that's what the original code was doing. Can go back later and clean it up.